controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -121,14 +122,32 @@ type Workload struct {
 var _ WorkloadCluster = &Workload{}
 
 func (w *Workload) getControlPlaneNodes(ctx context.Context) (*corev1.NodeList, error) {
-	nodes := &corev1.NodeList{}
-	labels := map[string]string{
-		labelNodeRoleControlPlane: "",
-	}
-	if err := w.Client.List(ctx, nodes, ctrlclient.MatchingLabels(labels)); err != nil {
```

Comment:
Have you considered doing two calls here to list and filter them by each label instead of querying all the nodes?

Reply:
Sounds fine to me.

So we would:
* list once with the old and new label
* merge the results and deduplicate

Would you keep the pagination? (I assume it's not strictly necessary as Kubernetes control plane nodes shouldn't be close to the default limit of the list call)

Reply:
Probably remove the pagination? In the future if we need to use limits we could have it back

Reply:
Thx for the review, good finding.

I've updated the code and the unit test accordingly